client/packages/invoices/src/OutboundShipment/DetailView/AddFromScannerButton.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect } from 'react';
+import {
+  useTranslation,
+  useToggle,
+  InvoiceNodeStatus,
+  useBarcodeScannerContext,
+  CircularProgress,
+  ScanIcon,
+  ScanResult,
+  ButtonWithIcon,
+  useNotification,
+} from '@openmsupply-client/common';
+import {
+  ItemRowFragment,
+  MasterListSearchModal,
+} from '@openmsupply-client/system';
+import { useOutbound } from '../api';
+
+export const AddFromScannerButtonComponent = ({
+  onAddItem,
+}: {
+  onAddItem: (item?: ItemRowFragment) => void;
+}) => {
+  const t = useTranslation('distribution');
+  const { status } = useOutbound.document.fields(['status']);
+  const isDisabled = status !== InvoiceNodeStatus.New;
+  const { addFromMasterList } = useOutbound.utils.addFromMasterList();
+  const { mutateAsync: getBarcodes } = useOutbound.utils.barcodes();
+  const { otherPartyId } = useOutbound.document.fields(['otherPartyId']);
+  const modalController = useToggle();
+  const filterByName = { existsForNameId: { equalTo: otherPartyId } };
+  const { hasBarcodeScanner, isScanning, startScanning, stopScan } =
+    useBarcodeScannerContext();
+  const { warning } = useNotification();
+
+  const handleScanResult = async (result: ScanResult) => {
+    if (!!result.content) {
+      const { content, gtin } = result;
+      const barcode = gtin ?? content;
+      const barcodes = await getBarcodes(barcode);
+
+      if (barcodes.totalCount === 0) {
+        warning(t('error.no-matching-item'))();
+        // TODO: save barcode after selection
+        onAddItem();
+        return;
+      }
+
+      onAddItem({ id: barcodes.nodes[0]?.itemId } as ItemRowFragment);
```

Comment:
If the intention is for a one to one match between a `barcode` and a `barcode:value` I wonder if it would be nicer to had a `getBarcode` endpoint, rather than using the filters and `getBarcodes` endpoint in this context.
Obviously a bit more work so I'm ok with how things are. Just think that would communicate more clearly the expectation.

Reply:
true. I was wondering at one point if we'd have matching items come back, along with pack sizes. We could then do something with the pack size info.. now that I've done the rest of it.. we aren't using the pack sizes anyway

Reply:
I think eventually barcodes API would be useful, but yeah I agree, with James, I like the idea of consumer driven API development, i.e. first do the front end with mock API, then fill API with functionality

Reply:
on the other hand, if we only ever implement the bare minimum which is closely aligned to the current implementation of the front end then we commit ourselves to api development for any future change. The advantage of graphQL is partly that we have features available and can quickly modify the f/e when implementing functionality.

Given that this is the consenus, I'll strip the functionality out: added to #1629

Reply:
> The advantage of graphQL is partly that we have features available

This can also be done with rest, not sure it's specific to graphQL

> if we only ever implement the bare minimum which is closely aligned to the current implementation of the front end then we commit ourselves to api development for any future change

Why do we `commit` ? Api can be changed and extended, and with shared types between front and back end so it's safe to do. 

I am still very strong about API based on current consumer requirements (rather then guessing what requirements should be and implementing something that prioritises future requirements vs ease of consumption of current requirements)